core/src/test/java/org/elasticsearch/index/IndexingSlowLogTests.java

```diff
@@ -62,6 +64,44 @@ public void testSlowLogParsedDocumentPrinterSourceToLog() throws IOException {
         p = new SlowLogParsedDocumentPrinter(index, pd, 10, true, 3);
         assertThat(p.toString(), containsString("source[{\"f]"));
         assertThat(p.toString(), startsWith("[foo/123] took"));
+
+        source = new BytesReference() {
```

Comment:
Is it possible to not implement BytesReference just to throw the exception but to actually use a "source" that will trigger some JsonParseException when creating the parser from source? I think that would be more realistic and a better test.